functorch/_src/decompositions.py

```diff
@@ -285,6 +285,55 @@ def huber_loss_backward(grad_output: Tensor, self: Tensor, target: Tensor, reduc
     )
 
 
+@register_decomposition(aten.nll_loss_backward)
+def nll_loss_backward(grad_output: Tensor, self: Tensor, target: Tensor, weight: Optional[Tensor], reduction: int, ignore_index: int, total_weight: Tensor) -> Tensor:
+    if self.dim() < 0 or self.dim() > 2:
```

Comment:
maybe just makes these asserts?

Reply:
In the future we might want to trace these into graphs, and asserts are easier to trace than control flow + raise errors :)

Reply:
Changed